bugsnag-android-core/src/androidTest/java/com/bugsnag/android/BugsnagStoreMigratorTest.kt

```diff
@@ -0,0 +1,125 @@
+package com.bugsnag.android
+
+import com.bugsnag.android.internal.BugsnagStoreMigrator
+import org.junit.After
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+import java.util.UUID
+
+class BugsnagStoreMigratorTest {
+    private var bugsnagStoreMigrator = BugsnagStoreMigrator
+    private lateinit var tmpdir: File
+
+    @Before
+    fun setUp() {
+        tmpdir = File(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString())
+        tmpdir.mkdirs()
+    }
+
+    @After
+    fun tearDown() {
+        tmpdir.deleteRecursively()
+    }
+
+    @Test
+    fun tesMoveFilesToNewDir() {
+        val filesToMove = listOf(
+            "bugsnag-sessions" to "bugsnag/sessions",
+            "last-run-info" to "bugsnag/last-run-info",
+            "user-info" to "bugsnag/user-info",
+            "bugsnag-native" to "bugsnag/native",
+            "bugsnag-errors" to "bugsnag/errors"
+        )
+        filesToMove.forEach { (from, to) ->
+            val file = File(tmpdir, from)
+            val newDir = File(tmpdir, to)
+            file.mkdirs()
+            bugsnagStoreMigrator.moveToNewDirectory(tmpdir)
+            assertFalse(file.isDirectory)
+            assertFalse(file.exists())
+            assertTrue(newDir.isDirectory)
+            assertTrue(newDir.exists())
+        }
+    }
+
+    @Test
+    fun testMoveOneFileToNewDirectory() {
+        val file = File(tmpdir, "bugsnag-native")
+        val newDirFile = File(tmpdir, "bugsnag/native")
+        file.mkdirs()
+        bugsnagStoreMigrator.moveToNewDirectory(tmpdir)
+        assertFalse(file.isDirectory)
+        assertFalse(file.exists())
+        assertTrue(newDirFile.exists())
+        assertTrue(newDirFile.isDirectory)
+    }
+
+    @Test
+    fun testNotToMoveUndefinedFile() {
+        val newDirFile = File(tmpdir, "bugsnag/test")
+        val file = File(tmpdir, "test")
+        file.mkdirs()
+        file.createNewFile()
+        assertTrue(file.isDirectory)
+        assertTrue(file.exists())
+        bugsnagStoreMigrator.moveToNewDirectory(tmpdir)
+        assertFalse(newDirFile.isDirectory)
+        assertFalse(newDirFile.exists())
+        assertTrue(file.isDirectory)
+        assertTrue(file.exists())
+    }
+
+    @Test
+    fun testDeepPathUndefinedFile() {
+        val file = File(tmpdir, "test/tes2/test3")
+        file.mkdirs()
+        file.createNewFile()
```

Comment:
The `creatNewFile` doesn't make sense in this test.

```suggestion
        val file = File(tmpdir, "test/tes2/test3").apply { mkdirs() }
```